components/security-mgt/org.wso2.carbon.security.mgt/src/main/java/org/wso2/carbon/security/keystore/KeyStoreManagementServiceImpl.java

```diff
@@ -180,12 +217,262 @@ public void addCertificate(String tenantDomain, String alias, String certificate
     public void deleteCertificate(String tenantDomain, String alias) throws KeyStoreManagementException {
 
         try {
-            getKeyStoreAdmin(tenantDomain).removeCertFromStore(alias, getKeyStoreName(tenantDomain));
+            KeyStoreAdmin keyStoreAdmin = getKeyStoreAdmin(tenantDomain);
+            String keystoreName = getKeyStoreName(tenantDomain);
+            if (!isPrivateKey(keyStoreAdmin, keystoreName, alias)) {
+                keyStoreAdmin.removeCertFromStore(alias, keystoreName);
+            }
+        } catch (Exception e) {
+            throw handleServerException(ERROR_CODE_DELETE_CERTIFICATE, alias, e);
+        }
+    }
+
+    public void addPrivateKey(String alias, String privateKeyContent, String certificateChain, String tenantDomain) throws KeyStoreManagementException {
```

Comment:
Method comment please

Reply:
added override annotation and added method level comment in the interface